### DISCUSSION TITLE
Vfd queue full?

### DISCUSSION THREAD

**Discussion by Monberg75** [winder/Universal-G-Code-Sender#2547] [Help]
I get this error sometimes, resulting in the machine crashing. As the spindle doesn't turn on.

Is this a UGS or machine/controller problem? And could UGS stop the program when this error occurs?

Xprov5 controller with fluidnc.

**Comment by breiler** [comment]
I'd say that this is a controller problem. If the controller gets an indication that the spindle could not turn on it should notify this with an alarm to the sender software.

**Comment by Monberg75** [comment]
Ok, it is shown as info in the ugs console and not a warning.

fre. 14. jun. 2024 07.49 skrev Joacim Breiler ***@***.***>:

**Comment by Monberg75** [comment]
Maybe I should say, that I always get this error starting a job. Never in
the middle of one, always on start. If that makes any difference.

I've also asked on sparks GitHub to see if they can help, maybe change the
message to a warning instead of an info message.

fre. 14. jun. 2024 09.12 skrev Monberg75 ***@***.***>:

**Reply by breiler** [reply]
This should not just result in just a INFO/WARN message, the controller should recognize this as a critical error and halt all motion and throw an alarm. I don't think that this special case should be implemented in all senders out there. 

Even if I were to implement this in UGS, I'd need the hardware to recreate this specific scenario.

**Comment by Monberg75** [comment]
Agreed, but as I don't know how sender and controller works together, I
didn't know where the problem was.

I will go to sparks, and ask them to update, as this is wrong.

Do you know if this is a controller problem or fluidnc problem then? Maybe
fluidnc is reporting the problem wrong?

fre. 14. jun. 2024 10.18 skrev Joacim Breiler ***@***.***>:

**Reply by breiler** [reply]
This is a firmware thing and it should be implemented in FluidNC.
